### PR TITLE
End screen sharing screen

### PR DESCRIPTION
**Pull request by EgorovEI**
Added module for end screen sharing screen. Presenting this screen will be added in separate PR.
Added unit tests for viewModel.
Snapshot tests will be added in separate task.

[MOB-1770](https://glia.atlassian.net/browse/MOB-1770)

[MOB-1770]: https://glia.atlassian.net/browse/MOB-1770?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: GliaWidgets/CallVisualizer/ScreenSharing/Coordinator/ScreenSharingCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+extension CallVisualizer {
+    final class ScreenSharingCoordinator: FlowCoordinator {
+        typealias ViewController = UIViewController
+        var delegate: ((DelegateEvent) -> Void)?
+
+        private let theme: Theme
+        private let environment: Environment
+        private var viewModel: ScreenSharingViewModel?
+        private var viewController: ScreenSharingViewController?
```
Comment (igorkravchenko) [inline GliaWidgets/CallVisualizer/ScreenSharing/Coordinator/ScreenSharingCoordinator.swift:11]: I think it is fine to omit it reference to view controller here.

File: GliaWidgets/CallVisualizer/ScreenSharing/Coordinator/ScreenSharingCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        private func showEndScreenSharingViewController() -> ViewController {
+            typealias Props = ScreenSharingViewController.Props
+
+            let viewController = ScreenSharingViewController()
+            viewController.modalPresentationStyle = .overFullScreen
+
+            let viewModel = ScreenSharingViewModel(
+                view: viewController,
+                style: theme.screenSharing,
+                output: viewModelOutput,
```
Comment (igorkravchenko) [inline GliaWidgets/CallVisualizer/ScreenSharing/Coordinator/ScreenSharingCoordinator.swift:40]: For consistency I think it is worth to stick to props assignment via delegate event. If there's more optimal approach, it makes sense to implement it as separate PR.

File: GliaWidgets/CallVisualizer/ScreenSharing/Coordinator/ScreenSharingCoordinator.swift (hunk 1 above)
Comment (EgorovEI) [reply GliaWidgets/CallVisualizer/ScreenSharing/Coordinator/ScreenSharingCoordinator.swift]: a6dd3c3de3c9d0a64905b3c0de1f4f5242ad53c1

File: GliaWidgets/CallVisualizer/ScreenSharing/Coordinator/ScreenSharingCoordinator.swift (hunk 2 above)
Comment (EgorovEI) [reply GliaWidgets/CallVisualizer/ScreenSharing/Coordinator/ScreenSharingCoordinator.swift]: a6dd3c3de3c9d0a64905b3c0de1f4f5242ad53c1

File: GliaWidgets/Sources/CallVisualizer/CallVisualizer.Coordinator.swift [hunk 3]
```diff
@@ -115,6 +112,30 @@ extension CallVisualizer {
...
+            let viewController = ScreenSharingViewController()
+            viewController.modalPresentationStyle = .overFullScreen
+
+            var viewModel: ScreenSharingViewModel? = ScreenSharingViewModel(
+                style: viewFactory.theme.screenSharing,
+                delegate: .init { [weak self, weak viewController] event in
+                    switch event {
+                    case .close:
+                        viewController?.dismiss(animated: true)
+//                        self?.delegate?(.close)
```
Comment (igorkravchenko) [inline GliaWidgets/Sources/CallVisualizer/CallVisualizer.Coordinator.swift:128]: Removable?

File: GliaWidgets/Sources/CallVisualizer/CallVisualizer.Coordinator.swift [hunk 4]
```diff
@@ -115,6 +112,30 @@ extension CallVisualizer {
...
+                    case let .propsUpdated(props):
+                        self?.end()
+                        break
+//                        viewController?.props = props
```
Comment (igorkravchenko) [inline GliaWidgets/Sources/CallVisualizer/CallVisualizer.Coordinator.swift:132]: Removable?

**Review by igorkravchenko — APPROVED** [review]
Looks good. Left a few minor comments.

File: GliaWidgets/Sources/CallVisualizer/CallVisualizer.Coordinator.swift (hunk 3 above)
Comment (EgorovEI) [reply GliaWidgets/Sources/CallVisualizer/CallVisualizer.Coordinator.swift]: Sorry, this is outdated.

File: GliaWidgets/Sources/CallVisualizer/CallVisualizer.Coordinator.swift (hunk 4 above)
Comment (EgorovEI) [reply GliaWidgets/Sources/CallVisualizer/CallVisualizer.Coordinator.swift]: Sorry, this is outdated.

**Review by rasmustautsglia — APPROVED** [review]

File: GliaWidgets/Sources/CallVisualizer/ScreenSharing/View/ScreenSharingView.swift [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            ) {
+                self.style = style
+                self.endScreenSharing = endScreenSharing
+                self.back = back
+            }
+        }
+
+        // MARK: - Properties
+
+        private lazy var header = Header(with: Theme().screenSharing.header).make { header in
```
Comment (igorkravchenko) [inline GliaWidgets/Sources/CallVisualizer/ScreenSharing/View/ScreenSharingView.swift:26]: I would suggest to pass styling via props to have single entry point for controlling view.

Comment (EgorovEI) [reply GliaWidgets/Sources/CallVisualizer/ScreenSharing/View/ScreenSharingView.swift:26]: Yeap, I guess I just forgot to pass it into entry point.

Comment (EgorovEI) [reply GliaWidgets/Sources/CallVisualizer/ScreenSharing/View/ScreenSharingView.swift]: [b1a8f14](https://github.com/salemove/ios-sdk-widgets/pull/470/commits/b1a8f14161e658948ceec99163d53c10ebe7778c)

**Comment by EgorovEI** [comment]
!squash

**Pull request merged**